zipline/assets/assets.py

```diff
@@ -529,33 +529,34 @@ def _lookup_generic_scalar(self,
         On success, append to matches.
         On failure, append to missing.
         """
-        try:
-            if isinstance(asset_convertible, Asset):
-                matches.append(asset_convertible)
+        if isinstance(asset_convertible, Asset):
+            matches.append(asset_convertible)
 
-            elif isinstance(asset_convertible, Integral):
+        elif isinstance(asset_convertible, Integral):
+            try:
                 result = self.retrieve_asset(int(asset_convertible))
-                if result is None:
-                    raise SymbolNotFound(symbol=asset_convertible)
-                matches.append(result)
+            except SidNotFound:
+                missing.append(asset_convertible)
+                return None
+            matches.append(result)
 
-            elif isinstance(asset_convertible, string_types):
-                # Throws SymbolNotFound on failure to match.
+        elif isinstance(asset_convertible, string_types):
+            # Throws SymbolNotFound on failure to match.
```

Comment:
This comment probably isn't necessary anymore given that it's in a try/catch block catching `SymbolNotFound`.